keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/BoundsCalculator.java

```diff
@@ -0,0 +1,75 @@
+package io.improbable.keanu.algorithms.variational;
+
+import io.improbable.keanu.tensor.TensorShape;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.Vertex;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.math3.optim.SimpleBounds;
```

Comment:
There was a previous comment mentioning that we shouldn't depend explicitly on this in Keanu - have we decided that it's ok to do that now?

Reply:
Good point. This class shouldn't be part of the public API and should be made package private and therefore just an implementation detail.